### PR TITLE
Created legend

### PR DESCRIPTION
**Pull request by mayavera**

### CODE REVIEW

File: src/ticketmap/index.tsx [hunk 1]
```diff
@@ -97,6 +100,48 @@ const $availableTicketGroups = createDeepEqualSelector(
     }).filter(ticketGroup => ticketGroup)
 )
 
+const $priceSortedTicketGroups = createDeepEqualSelector(
+  $ticketGroups,
+  ticketGroups => ticketGroups.sort((a, b) => a.retail_price - b.retail_price)
+)
+
+const $sectionPercentiles = (state: State, props: Props) => props.sectionPercentiles
```
Comment (josephspens) [inline src/ticketmap/index.tsx]: Why does this have state _and_ props params?

Comment (mayavera) [reply src/ticketmap/index.tsx]: This function signature is required for selectors that access props. It allows other selectors to compose this one without fiddling with parameter order. For example:

```js
import { createSelector } from 'reselect'

const $sectionPercentiles = (state: State, props: Props) => props.sectionPercentiles

const $selectedPercentile = createSelector(
  $sectionPercentiles,
  (state: State) => state.selectedPercentileId,
  (percentiles, id) => percentiles[id]
)

/* ... */

render() {
  return <div>{$selectedPercentile(state, props)}</div>
}
```

**Comment by mayavera** [comment]
> This change is adding a lot of reselect logic, which is duplicating the existing percentile logic in `getDefaultColor`. Can you dry them out?

Sur thing!

File: src/ticketmap/index.tsx (hunk 1 above)
Comment (josephspens) [reply src/ticketmap/index.tsx]: Ah I see, the selector params are passed into each of the embedded functions (except for the last one) in the same order as they were passed into the selector. The first function needs props, and the second function needs state, so the second function can have a signature without the second param but the first function can't have a signature without the first.

Comment (mayavera) [reply src/ticketmap/index.tsx]: Spot on!

File: src/ticketmap/index.tsx [hunk 2]
```diff
@@ -448,13 +501,16 @@ export default class TicketMap extends Component<Props, State> {
...
+
+    const ranges = $costRanges(this.state, this.props)
+
+    for (const range of ranges) {
+      if (range.max > lowestTicketPriceInSection) {
+        return range.color
       }
     }
+
+    return ranges[ranges.length - 1]
```
Comment (josephspens) [inline src/ticketmap/index.tsx]: I don't believe this is the same return value as `range.color`. Might be helpful to add types to this function (both method signature and the code within it).

File: src/ticketmap/index.tsx [hunk 3]
```diff
@@ -448,13 +501,16 @@ export default class TicketMap extends Component<Props, State> {
   getDefaultColor(ticketGroups: TicketGroup[]) {
     const { sectionPercentiles } = this.props
```
Comment (josephspens) [inline src/ticketmap/index.tsx]: I don't think this is needed anymore.

**Pull request merged**